[CI/Build]: Add nightly build container image of latest code

hickeyma opened the pull request:
Workflow to build nightly container image of latest code of LMcache and vLLM. Pushes the image to [Dockerhub](https://hub.docker.com/r/lmcache/vllm-openai).

**Note:** Encountered issues using [docker/build-push-action](https://github.com/docker/build-push-action) GithHub action when building the image as it could not resolve dynamic versioning in LMCache. Therefore, calling the docker `build` and `push` commands directly in steps.

Partial #594

~~Note to reviewers:Do NOT merge until Push/Pull events are removed (for testing purpose during PR as its on schedule):~~ REMOVED

hickeyma committed:
Add nightly build container image of latest code

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

sammshen commented:
> Note to reviewers: Do NOT merge until Push/Pull events are removed (for testing purpose during PR as its on schedule):

Sorry if I'm misunderstanding @hickeyma but do we not need to merge your pipeline to dev before we can see it in action? And then we should do a quick follow up PR removing the Push/PR triggers? 

Alternatively, we could debug/test the pipeline by triggering on PR / push to a non-dev branch? (e.g. daily-docker-debug)

looks ready to merge / test out the pipeline to me!!

hickeyma commented:
>Sorry if I'm misunderstanding @hickeyma but do we not need to merge your pipeline to dev before we can see it in action? And then we should do a quick follow up PR removing the Push/PR triggers?

Yes but remove the Push/PR triggers first.

hickeyma commented:
~~Making the PR a draft for the moment as hit an issue where  Docker does not seem to be copying `.git`  into the container when building images. Because of this, the version inference is failing. See [setuptools-scm](https://setuptools-scm.readthedocs.io/en/latest/usage/#with-dockerpodman) for more details.~~

~~Trying to figure out how to get this to work.~~

Dropped [docker/build-push-action](https://github.com/docker/build-push-action) GithHub action for now and calling docker commands doirectly in steps.

hickeyma committed:
Build and push container image directly

Encountered issues using docker/build-push-action GithHub action
to build image as it could not resolve dynamic versioning in LMCache.

Therefore, calling the docker build and push commands directly.

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented:
Workflow now running and published images to DockerHub:

![Screenshot 2025-06-05 at 10 58 40](https://github.com/user-attachments/assets/be11930f-94b6-494e-b556-ba04c40ca6ec)

ApostaC reviewed (approved):
LGTM!

hickeyma committed:
Remove push/pull events

Events just to show that workflow runs. It is a scheduled event,
so workflow wouldn't run normally on commits.

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma committed:
Update build args after Dockerfile update

Dockerfile changes in PR #746 requires updates to build command.

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

sammshen commented:
LGTM! This is really nice!
